TypeCobol.LanguageServer/Workspace.cs

```diff
@@ -573,20 +563,18 @@ public void DidChangeConfigurationParams(string[] arguments)
                 analyzerProviderWrapper.AddProvider(a);
             }
 
-            CompilationProject = new CompilationProject(_workspaceName, _rootDirectoryFullName, Helpers.DEFAULT_EXTENSIONS, Configuration.Format, typeCobolOptions, analyzerProviderWrapper);
+            // DidChangeConfiguration notification is always for the Default Workspace Project.
+            // The is for compatibility with for instance TypeCobol plugin and fallback.
+            this.WorkspaceProjectStore.DefaultWorkspaceProject.ReplaceCompilationProject(
+                Configuration, typeCobolOptions, analyzerProviderWrapper);
 
-            if (Configuration.CopyFolders != null && Configuration.CopyFolders.Count > 0)
-            {
-                foreach (var copyFolder in Configuration.CopyFolders)
-                {
-                    CompilationProject.SourceFileProvider.AddLocalDirectoryLibrary(copyFolder, false,
-                        new[] {".cpy"}, Configuration.Format.Encoding,
-                        Configuration.Format.EndOfLineDelimiter, Configuration.Format.FixedLineLength);
-                }                
-            }
+            // Update All WorkspaceProject instances because global configurations have change
+            this.WorkspaceProjectStore.UpdateProjectOptionsAndAnalyzerProvider(typeCobolOptions, analyzerProviderWrapper);
```

Comment:
The Format changes are lost here.

Upon receiving a DidChangeConfigurationNotification, 4 things can change in a CompilationProject:
- format
- options
- analyzerprovider
- copyfolders

For the default project, all of them are taken in account because the CompilationProject is fully recreated, using the new values. But for named projects, format is omitted.

I think we should have a single method to update/modify a WorkspaceProject and its associated CompilationProject. Something like `WorkspaceProject.Update(DocumentFormat format, TypeCobolOptions options, AnalyzerProviderWrapper analyzerProviderWrapper, List<string> copyFolders)`. As for the implementation, we can either recreate the project or set its properties. This update method would also be responsible for clearing copy cache if need be andd scheduling file refresh.